entities/door/door.gd

```diff
@@ -0,0 +1,50 @@
+extends StaticBody2D
+
+var is_interactible = false
+var is_open = false
+
+func _ready() -> void:
+	$Open_Door_Sprite2D.visible=false
+	$Timer.wait_time = $CPUParticles2D.lifetime
+
+func _on_area_2d_body_entered(body: Node2D) -> void:
+	is_interactible = true
+	print("Press interact(E) to open or click to destroy")
+
+
+func _on_area_2d_body_exited(body: Node2D) -> void:
+	is_interactible = false
+
+
+func _input(event: InputEvent) -> void:
```

Comment:
Do we need another button? Can we just open the door when the player tries to walk into ie (ie collides with it)?

Fewer controls the better...

Plus, adding an extra button mean more headaches for making it work on mobile

Reply:
I don't know, I can certainly remove it if you prefer.